index.js

```diff
@@ -1,6 +1,20 @@
 const WHITE = [255, 255, 255, 1]
 const BLACK = [0, 0, 0, 1]
 
+// Cache for CSS variable values to avoid repeated getComputedStyle calls
+let cssVariableCache = new Map()
+
+/**
+ * Get a CSS variable value with caching
+ */
+const getCSSVariable = (cssVar) => {
+  if (!cssVariableCache.has(cssVar)) {
+    const value = window.getComputedStyle(document.body).getPropertyValue(cssVar).trim()
+    cssVariableCache.set(cssVar, value || '')
```

Comment:
@coderabbitai it looks like this could possibly be a bug if the correct value for the CSS variable is an empty String. Is there some other placeholder value we could use to indicate that we've already computed the value in the cache? Maybe we can make our own that can be guaranteed to be unique, like a variable with a dummy value?